src/main/java/com/atharvadholakia/password_manager/service/CredentialService.java

```diff
@@ -47,17 +53,21 @@ public Credential updateCredential(
 
   public Credential getCredentialById(String id) {
     log.info("Calling repository from service to get a credential with ID: {}", id);
-    return credentialRepository
-        .findById(id)
-        .orElseThrow(
-            () -> {
-              return new ResourceNotFoundException("Credential not found with ID " + id);
-            });
+    Credential credential =
+        credentialRepository
+            .findById(id)
+            .orElseThrow(() -> new ResourceNotFoundException("Credential not found with ID " + id));
+
+    credential.setPassword(encryptionService.decrypt(credential.getPassword()));
+    return credential;
   }
 
   public List<Credential> getAllCredentials() {
     log.info("Calling repository to get all the credentials");
-    return credentialRepository.findAll();
+    List<Credential> credentials = credentialRepository.findAll();
+    credentials.forEach(
+        credential -> credential.setPassword(encryptionService.decrypt(credential.getPassword())));
```

Comment:
I have decrypted all the passwords when we retrieve all the credentials(also when we retrieve a single credential by specific ID). Do we return encrypted password or decrypted when all the credentials are retrieved or a single credential is retrieved by ID?